examples_test.go

```diff
@@ -67,3 +67,25 @@ func ExampleScan_parseDuration() {
 	// Output:
 	// 3m20s
 }
+
+func ExampleScan_repeatedly() {
+	line := []byte("www.google.com:1234 www.google.com:2345")
+	r := regexp.MustCompile(`(\S+):(\d+)`)
+
+	for {
+		var (
+			position re.Position
```

Comment:
Looking at this in context, perhaps re.Range would be a better name (Position implies a single int)

Reply:
I've adopted this change, but lowercase `range` is a reserved word in Go, so `var range Range` isn't going to fly. Throughtout my code, I've been using `rng` to deal with this, but if that's not ideal, we might reconsider naming again.

Reply:
I chose re.Span.